### PR TITLE
Build refactoring and updating final plugins...

### PR DESCRIPTION
**Pull request by hntd187**
I reset my master so it's the same request as before for #438 

Okay, this should be just normal commits and not 900 extra commits I somehow put in there.

This request looks huge, but it's really not...

Added a shell script to test both JDKs
Did all the build plugins to latest versions and corrected problems
Refactored versions and exclusion rules into their own files so they can be included and managed easier than the dependency files.
Fixed some warnings
Because a scalariform style default was deprecated the plugin chose to reformat literally the entire code base, so that's why we have this. I can honestly understand not wanting this at all, so I can back it, just lemme know, but it is doing what the plugin was told to do, so it's not unexpected.

### CODE REVIEW

**Comment by hntd187** [comment]
Flaky test is flaky, can we retest @velvia ?

**Comment by noorul** [comment]
Restarted

**Comment by velvia** [comment]
Thanks for all your help @noorul

**Comment by noorul** [comment]
Pleasure

**Comment by noorul** [comment]
I see that you are not pinning most of the dependencies. Is that a good idea?

**Comment by hntd187** [comment]
@noorul what do you mean? All the versions are being locked  at specific versions,  some of them are ancient which lead to this upgrade.

**Comment by noorul** [comment]
@hntd187 My bad, I overlooked those variables to which version is assigned.

File: project/Dependencies.scala [hunk 1]
```diff
@@ -1,45 +1,35 @@
...
-    "com.typesafe.akka" %% "akka-slf4j" % "2.3.15" % "provided",
-    "com.typesafe.akka" %% "akka-cluster" % "2.3.15" exclude("com.typesafe.akka", "akka-remote"),
-    "io.spray" %% "spray-json" % "1.3.2",
-    "io.spray" %% "spray-can" % "1.3.3",
-    "io.spray" %% "spray-caching" % "1.3.3",
-    "io.spray" %% "spray-routing" % "1.3.3",
-    "io.spray" %% "spray-client" % "1.3.3",
+    "com.typesafe.akka" %% "akka-slf4j" % akka % "provided",
+    "com.typesafe.akka" %% "akka-cluster" % akka exclude("com.typesafe.akka", "akka-remote"),
+    "io.spray" %% "spray-json" % sprayJson,
```
Comment (velvia) [inline project/Dependencies.scala]: Not a biggie, but I wonder if `Version.sprayJson` etc, would be more readable.  Probably not.

File: scalastyle-config.xml [hunk 2]
```diff
@@ -19,7 +19,7 @@
  <check level="error" class="org.scalastyle.scalariform.SpacesBeforePlusChecker" enabled="true"></check>
  <check level="error" class="org.scalastyle.file.FileLineLengthChecker" enabled="true">
   <parameters>
-   <parameter name="maxLineLength"><![CDATA[110]]></parameter>
+   <parameter name="maxLineLength"><![CDATA[180]]></parameter>
```
Comment (velvia) [inline scalastyle-config.xml]: Okay, I disagree with this one.  I hate hate long runaway lines!   Code is far more readable when lines are shorter.  Though I'm sure you know this and there's some reason why this is being increased.

**Comment by velvia** [comment]
@hntd187 looked at first three commits, minor comments, well ok one about line length I'd like changed.

**Comment by hntd187** [comment]
Well personally, I use some large wide screen monitors so short lines drive me kinda nuts, I read code very left to right and as long as it's not like 250 characters long I prefer a longer line, but that's just me personally, I'm sure there would be a lot of disagreement over that sentiment :-) Either way, I will change it back. @velvia

**Comment by noorul** [comment]
LGTM

**Comment by hntd187** [comment]
@noorul @velvia please take note with this that I disabled scalariform for the build because the new version will reformat all the code, why I am not sure because the settings are the same from version to version, so it seems more on scalariform changing something. I'll submit a pull request with it turned back on so we can see the differences.

**Comment by noorul** [comment]
ack

**Pull request merged**